Assets/Scripts/LevelSelect/BountySystem/BountySelect.cs

```diff
@@ -11,33 +12,53 @@
 public class BountySelect : MonoBehaviour
 {
     [SerializeField] private FadeScreenHandler fadeScreen;
-    [SerializeField] private RectTransform bountyGrid;
+    [SerializeField] private Transform[] bountyGrid;
     [SerializeField] private BountyButton bountyButtonPrefab;
+    [SerializeField] private BountyRewardIconDatabase bountyRewardIconDatabase;
+
+
+    [SerializeField] private TMP_Text bountyTitle;
+    [SerializeField] private TMP_Text bountyDescription;
+    [SerializeField] private TMP_Text bountyRewards;
+    [SerializeField] private TMP_Text chooseBountyText;
 
 #nullable enable
     protected virtual void Awake()
     {
+        BountyButton.BountyOnHoverEvent += OnBountyHover;
+        BountyButton.BountyOnHoverEndEvent += OnBountyHoverEnd;
         ConstructBountyButtons();
         fadeScreen.SetDarkScreen();
         StartCoroutine(fadeScreen.FadeInLightScreen(1f));
     }
 
+    void OnDestroy()
+    {
+        BountyButton.BountyOnHoverEvent -= OnBountyHover;
+        BountyButton.BountyOnHoverEndEvent -= OnBountyHoverEnd;
+    }
+
     public void ConstructBountyButtons()
     {
-        var bountyCollection = BountyManager.Instance.SelectedBountyInformation?.BountyCollection;
+        var bountyCollection = IBounties.Values.SelectMany(innerList => innerList).ToList();
```

Comment:
I'm not too sure about the flow of how bounties work. I thought you enter the contract scene, select a bounty, and then hit play to start that bounty (switching the scene to the respective level, etc)? I'm not sure what the buttons below the level select do. 

IBounties.Values.ToList() does not work as IBounties.Values is type IEnumerable<IEnumerable<IBounties>>. I need to fold the enumerables into one single list.

Reply:
Oh okay, you are correct with your interpretation, i think the confusion comes from the name of the field. 

So right now if you go to level select, you should be able to access bounties, but different stages are supposed to have different sets of bounties. (Not true rn though even though there are 4 existing entry points). So depending on which stage's bounty you select it shows a different SET of bounties (e.g. princessFrogBountyies or a new set of queenBeetleBounties).

Thats what that line of code above does, instead of rendering all possible bounties that exist, it only renders bounties for the specific set that was chosen during level select.